rct229/rulesets/ashrae9012019/section19/section19rule17.py

```diff
@@ -60,24 +61,36 @@ def is_applicable(self, context, data=None):
 
     def create_data(self, context, data):
         rmi_b = context.baseline
-        dict_of_zones_and_terminal_units_served_by_hvac_sys = (
-            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi_b)
-        )
-
-        zone_data_b = {
-            zone["id"]: zone
-            for zone in find_all("$.buildings[*].building_segments[*].zones[*]", rmi_b)
-        }
 
         return {
-            "dict_of_zones_and_terminal_units_served_by_hvac_sys": dict_of_zones_and_terminal_units_served_by_hvac_sys,
-            "zone_data_b": zone_data_b,
+            "dict_of_zones_and_terminal_units_served_by_hvac_sys": (
+                get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi_b)
+            ),
+            "zone_data_b": {
+                zone["id"]: zone
+                for zone in find_all(
+                    "$.buildings[*].building_segments[*].zones[*]", rmi_b
+                )
```

Comment:
I prefer having data passed through instead of objects (nested dictionary). However that is personal preference, and I will leave it up to you.

The zones saved in the `zone_data_b` are used for calculating the zone exhaust fan powers.
If just data, I will do:
```python
zone_exhaust_fan_power_dict_b: {
    zone["id"]: sum([
        get_fan_object_electric_power(exhaust_fan )
        for exhaust_fan in find_all("$.zonal_exhaust_fan", zone)
    ])
    for zone in find_all(
        "$.buildings[*].building_segments[*].zones[*]", rmi_b
    )
}
```

Reply:
Good point and I like this approach. Addressed.